src/modules/map/components/CompareSlider.vue

```diff
@@ -10,27 +10,65 @@ const props = defineProps({
         default: window.innerWidth,
     },
 })
-
+const getMap = inject('getMap')
 const { clientWidth } = toRefs(props)
+const olMap = ref(null)
```

Comment:
Do not put the map in a ref, that will wrap it around a Vue Reactivity proxy.
As the Map instance is quite big (and has a lot of attributes/functions) this will be very CPU intensive, and isn't needed.

Whenever you need the map instance, call the inject function and checks that you have a valid instance before doing stuff

If you need to keep track of things you have already linked to an event, a binding function typically returns an identifier, or callback function, so that you may remove it later (check https://openlayers.org/en/latest/apidoc/module-ol_Map-Map.html#on )